PlayCover/Views/App Views/StoreAppView.swift

```diff
@@ -67,33 +66,64 @@ struct StoreAppConditionalView: View {
                     .padding(.horizontal, 5)
                     .foregroundColor(.secondary)
             }
+            .task {
+                iconUrl = try? await getIconURLFromBundleIdentifier(app.id, app.region)
+            }
         } else {
             VStack(alignment: .center, spacing: 0) {
-                AsyncImage(
-                    url: URL(string: app.icon),
-                    content: { image in
+                VStack {
+                    AsyncImage(url: iconUrl) { image in
                         image
                             .resizable()
-                    },
-                    placeholder: {
+                            .aspectRatio(contentMode: .fit)
+                    } placeholder: {
                         ProgressView()
                             .progressViewStyle(.circular)
                     }
-                )
-                .frame(width: 88, height: 88)
-                .cornerRadius(10)
+                }
+                .cornerRadius(15)
+                .frame(width: 70, height: 70)
                 .shadow(radius: 1)
-                .padding(.top, 8)
+                .padding(.vertical, 5)
                 HStack {
                     Image(systemName: "arrow.down.circle")
                         .font(.system(size: 16))
                     Text(app.name)
                         .lineLimit(2)
                         .multilineTextAlignment(.center)
                 }
-                .padding(.bottom, 14)
+                .padding(.vertical, 5)
             }
             .frame(width: 150, height: 150)
+            .task {
+                iconUrl = try? await getIconURLFromBundleIdentifier(app.id, app.region)
+            }
+        }
+    }
+
+    func getIconURLFromBundleIdentifier(_ bundleIdentifier: String,
+                                        _ region: StoreAppData.Region) async throws -> URL? {
```

Comment:
You are using do - catch statement inside function. So no need to make function throws.

Reply:
URLSession can also throw errors, so the throw is necessary.

Reply:
Move them inside do-catch